ARGO-3702 Add list of available services types per tenant

kaggis opened the pull request:
# Goal 
provide a list of available types of services per tenant in the form
```json
[
  { "name": "database", "description": "a quick service type description"},
  { "name": "api", "description": "a quick service type description"}
]
```

# Implementations

### ⚡ Create a list of available service types for a specific date
Request ➡️  `HTTP` `POST` `/api/v2/topology/service-types`
`post_body:`
```
[
  { "name": "database", "description": "a quick description of service type: database"},
  { "name": "api", "description": "a quick description of service type: api"}
]
```
Response ⬅️ `201 OK`

```json
{
    "message": "Topology of 2 service types created for date: YYYY-MM-DD",
    "code": "201"
}
```
 
### ⚡ List available service types for a specific date
Request ➡️  `HTTP` `GET` `/api/v2/topology/service-types?date=YYYY-MM-DD`

Response ⬅️ `200 OK`

```json
{
  "status": {
    "message": "Success",
    "code": "200"
  },
  "data": [
    {
      "date": "YYYY-MM-DD",
      "name": "database",
      "description": "a quick description of service type: database"
    },
    {
      "date": "YYYY-MM-DD",
      "name": "api",
      "description": "a quick description of service-type: api"
    }
  ]
}
```

### ⚡ Delete list of available service types for a specific date
Request ➡️  `HTTP` `DELETE` `/api/v2/topology/service-types?date=YYYY-MM-DD`

Response ⬅️ `200 OK`
```
{
    "message": "Topology of 2 service types deleted for date: YYYY-MM-DD",
    "code": "200"
}
```

- [x] Added ServiceType Model in topology
- [x] Added CreateServiceTypes method in topology handlers
- [x] Added DeleteServiceTypes method in topology handlers
- [x] Added ListServiceTypes method in topology handlers
- [x] Configured and added new routes in topology sub-router
- [x] Added unit tests
- [x] Updated roles
- [x] Updated swagger
- [x] Updated docs

themiszamani commented:
The PR is perfect. The only thing that confuses me a little is the use of the word service as an attribute. 
 { "service": "database", "description": "a quick description of service: database"},
Maybe we could use something more generic like name. Let's discuss it.

kaggis commented:
Fixed definition from `service` to `service-type`
- `/api/v2/services` --> `/api/v2/service-types`
-  Fixed json representation of a service type item to: 
    ``` json{ "name": "database", "description": "a quick service type description"}```
